crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -1784,6 +1832,8 @@ mod test {
                 "fn f(f: pub Field, y : Field, z : comptime Field) -> u8 { x + a }",
                 "fn f<T>(f: pub Field, y : T, z : comptime Field) -> u8 { x + a }",
                 "fn func_name<T>(f: Field, y : T) where T: SomeTrait {}",
+                // The following should produce compile error on later stage. From the parser prespective it's fine
+                "fn func_name<A>(f: Field, y : pub Field, z : pub [u8;5]) where T: SomeTrait {}",
```

Comment:
I assume the expected compiler error in a later stage is the usage of `pub` here in a non-entry point function. I don't see why we need to test with this at all though, since it isn't valid input.

Reply:
Haha, No. The intent here is that the generics is named <A>, but the where clause applies to <T>

the pub is an other problem, so to make the test case clear to the reader I removed all the pubs in the parameter list